src/connection.ts

```diff
@@ -127,8 +127,14 @@ export class ConnectionManager implements IConnectionManager {
           this.logger.debug('OUTPUT: ' + data.toString());
 
           // If the language client isn't already running and it's sent the trigger text, start up a client
-          if (this.languageServerClient === undefined && data.toString().match('LANGUAGE SERVER RUNNING') !== null) {
-            this.languageServerClient = this.createLanguageClient();
+          if (this.languageServerClient === undefined && /LANGUAGE SERVER RUNNING/.test(data.toString())) {
+            if(this.connectionConfiguration.port){            
+              this.languageServerClient = this.createLanguageClient();
+            }else{
+              var p = data.toString().match(/LANGUAGE SERVER RUNNING \d+\.\d+\.\d+\.\d+:(\d+)/);
```

Comment:
The hostname can be text e.g. `localhost`. so the IP based regex will fail.  Perhaps

`/LANGUAGE SERVER RUNNING .+:(\d+)$/` ?

Reply:
Good point, and I wasn't even using that match. Though, I think its `.*` isn't it?